src/lib.rs

```diff
@@ -25,50 +29,84 @@ impl PyEvtxParser {
 
         obj.init({
             PyEvtxParser {
-                inner: inner.into_chunks(),
-                records: None,
-                settings: ParserSettings::new(),
+                inner: Some(inner)
             }
         });
 
         Ok(())
     }
+
+    fn records(&mut self) -> PyResult<PyRecordsIterator> {
+        self.records_iterator(OutputFormat::XML)
+    }
+
+    fn records_json(&mut self) -> PyResult<PyRecordsIterator> {
+        self.records_iterator(OutputFormat::JSON)
+    }
 }
 
 impl PyEvtxParser {
-    fn err_to_object(e: failure::Error, py: Python) -> PyObject {
-        PyErr::new::<RuntimeError, _>(format!("{}", e)).to_object(py)
+    fn records_iterator(&mut self, output_format: OutputFormat) -> PyResult<PyRecordsIterator> {
+        let inner = match self.inner.take() {
+            Some(inner) => inner,
+            None => {
+                return Err(PyErr::new::<RuntimeError, _>("PyEvtxParser can only be used once"));
+            }
+        };
+
+        Ok(PyRecordsIterator {
+            inner: inner.into_chunks(),
+            records: None,
+            settings: ParserSettings::new(),
+            output_format: output_format,
+        })
     }
+}
 
-    fn record_to_pydict(gil: Python, record: SerializedEvtxRecord) -> PyResult<&PyDict> {
-        let pyrecord = PyDict::new(gil);
 
-        pyrecord.set_item("event_record_id", record.event_record_id)?;
-        pyrecord.set_item("timestamp", format!("{}", record.timestamp))?;
-        pyrecord.set_item("data", record.data)?;
-        Ok(pyrecord)
-    }
+fn err_to_object(e: failure::Error, py: Python) -> PyObject {
+    PyErr::new::<RuntimeError, _>(format!("{}", e)).to_object(py)
```

Comment:
I think it would be really sweet if we could bin the errors from `evtx` to all the ones that are I/O related, and maybe raise an IOError in that case. A la https://github.com/BurntSushi/ripgrep/compare/ag/snafu#diff-62a73ebf355cee98e3a01bb067337e19R207, what do you think?

Reply:
Also, PyO3 implement `From<io::Error>` for `PyErr`, which will automatically return an python IO exception.